WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -183,7 +183,12 @@ def update
       end
     else
       flash.now[:danger] = I18n.t('registrations.flash.failed')
-      render :edit
+      if params[:from_admin_view]
+        render :edit
+      else
+        self.register # Set instance variables needed by :register view
```

Comment:
Yeah that's definitely some refactor that could be addressed in a later PR anyway!